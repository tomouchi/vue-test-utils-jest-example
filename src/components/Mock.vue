<template>
  <div>
    <div>{{ test1 }}</div>
    <div>{{ test2 }}</div>
    <div>{{ test3 }}</div>
  </div>
</template>

<script>
import axios from "axios";
export default {
  name: "Mock",
  data: function() {
    return {
      test1: null,
      test2: null,
      test3: null
    };
  },
  created: function() {
    axios
      .get("http://test1.test")
      .then(response => (this.test1 = response.data))
      .catch(() => {});
    axios
      .get("http://test2.test")
      .then(response => (this.test2 = response.data))
      .catch(() => {});
    axios
      .get("http://test3.test")
      .then(response => (this.test3 = response.data))
      .catch(() => {});
  }
};
</script>
