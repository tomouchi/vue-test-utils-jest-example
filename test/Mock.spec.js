import { shallowMount } from "@vue/test-utils";
import Mock from "@/components/Mock.vue";
import moxios from "moxios";

describe("Mock.vue", function() {
  beforeEach(function() {
    // import and pass your custom axios instance to this method
    moxios.install();
  });
  afterEach(function() {
    // import and pass your custom axios instance to this method
    moxios.uninstall();
  });
  it("just for a single spec", function(done) {
    moxios.withMock(function() {
      const responses = {
        "http://test1.test": { status: 200, response: "test1" },
        "http://test2.test": { status: 404 },
        "http://test3.test": { status: 200, response: "test3" }
      };

      let wrapper = shallowMount(Mock);

      moxios.wait(function() {
        respondAllWith(moxios, responses).then(() => {
          expect(wrapper.html()).toMatchSnapshot();
          done();
        });
      });
    });
  });
});

function respondAllWith(moxios, responses) {
  return moxios.requests.__items.reduce((accumulator, promise) => {
    return promise.respondWith(responses[promise.url]);
  }, null);
}
