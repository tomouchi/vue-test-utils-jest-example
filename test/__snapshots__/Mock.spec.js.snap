// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mock.vue just for a single spec 1`] = `
<div>
    <div>test1</div>
    <div></div>
    <div>test3</div>
</div>
`;
